api/main.py

```diff
@@ -63,6 +70,15 @@ async def create_indexes():
     """Startup event handler to create database indexes"""
     await db.create_indexes()
 
+@app.on_event('startup')
+async def fastapi_users_beanie_init():
+    """Startup event handler for beanie"""
+    await init_beanie(
+        database=db.db,  
+        document_models=[
+            TestUser,  
+        ],
+    )
```

Comment:
Maybe this call could be moved to `api.auth` with a method in the Database class to avoid exposing the underlying `.db` object?  That way only the Database class would follow the object-oriented principle of accessing its own data.  We already know we need to import the `beanie` modules in `api.auth` so it wouldn't be adding extra dependencies there.

Reply:
The reason I kept this call in `main.py` was that `init_beanie` needs an actual instantiated database object and not just the class reference. Still, I'll take another look if I can move things around.